core/core_test.go

```diff
@@ -101,6 +101,33 @@ func init() {
 	}
 }
 
+func TestIdempotentAppInitialization(t *testing.T) {
```

Comment:
nit: We don't want `New` to be idempotent and that's not really what this PR does. Idempotent means that subsequent calls are effectively a no-op. What we want is for subsequent calls to return a new app instance (and for closing the old instance to not leave any dangling goroutines).

Reply:
Yeah, I was thinking about that this morning. In general, idempotent means that the result of applying a function is the same after the first application. Admittedly, using that terminology here is a stretch.